ksqldb-engine/src/main/java/io/confluent/ksql/internal/PullQueryExecutorMetrics.java

```diff
@@ -620,38 +668,54 @@ private void addSensor(
 
   private Map<MetricsKey, Sensor> configureSensorMap(
       final String sensorBaseName, final MetricsAdder metricsAdder) {
-    final ImmutableMap.Builder<MetricsKey, Sensor> builder
-        = ImmutableMap.builder();
+    final ImmutableMap.Builder<MetricsKey, Sensor> builder = ImmutableMap.builder();
 
     for (final PullSourceType sourceType : PullSourceType.values()) {
       for (final PullPhysicalPlanType planType : PullPhysicalPlanType.values()) {
         for (final RoutingNodeType routingNodeType : RoutingNodeType.values()) {
-          final String variantName = sourceType.name().toLowerCase() + "-"
-              + planType.name().toLowerCase() + "-" + routingNodeType.name().toLowerCase();
-          final Sensor sensor = metrics.sensor(
-              PULL_QUERY_METRIC_GROUP + "-" + PULL_REQUESTS + "-" + sensorBaseName + "-"
-                  + variantName);
-
-          final ImmutableMap<String, String> tags = ImmutableMap.<String, String>builder()
-              .putAll(customMetricsTags)
-              .put(KsqlConstants.KSQL_QUERY_SOURCE_TAG, sourceType.name().toLowerCase())
-              .put(KsqlConstants.KSQL_QUERY_PLAN_TYPE_TAG, planType.name().toLowerCase())
-              .put(KsqlConstants.KSQL_QUERY_ROUTING_TYPE_TAG,
-                  routingNodeType.name().toLowerCase())
-              .build();
-
-          metricsAdder.addMetrics(sensor, tags, variantName);
-
-          builder.put(new MetricsKey(sourceType, planType, routingNodeType), sensor);
-          sensors.add(sensor);
+          addSensorToMap(
+              sensorBaseName,
+              metricsAdder,
+              builder,
+              new MetricsKey(sourceType, planType, routingNodeType)
+          );
         }
       }
     }
 
+    // Add one more sensor for collecting metrics when there is no response
+    addSensorToMap(sensorBaseName, metricsAdder, builder, new MetricsKey());
+
     return builder.build();
   }
 
+  private void addSensorToMap(final String sensorBaseName, final MetricsAdder metricsAdder,
+      final Builder<MetricsKey, Sensor> builder, final MetricsKey metricsKey) {
+    final String variantName = metricsKey.variantName();
+    final Sensor sensor = metrics.sensor(
+        PULL_QUERY_METRIC_GROUP + "-"
+            + PULL_REQUESTS + "-"
+            + sensorBaseName + "-"
+            + variantName);
+
+    final ImmutableMap<String, String> tags = ImmutableMap.<String, String>builder()
+        .putAll(customMetricsTags)
+        .put(KsqlConstants.KSQL_QUERY_SOURCE_TAG, metricsKey.sourceTypeName())
+        .put(KsqlConstants.KSQL_QUERY_PLAN_TYPE_TAG, metricsKey.planTypeName())
+        .put(KsqlConstants.KSQL_QUERY_ROUTING_TYPE_TAG, metricsKey.routingNodeTypeName())
```

Comment:
`.toLowerCase()` missing?

Reply:
It is now encapsulated inside the xYName methods.